### ISSUE TITLE
error message in selectVariables

### ISSUE THREAD

**Issue by MarthaBernal** [Macaulay2/M2#692]
I tried this on M2:
```
R=QQ[x,y,MonomialOrder=>Lex,Inverses=>true];
selectVariables({0},R)
```
and obtained as the second output:
stdio:12:1:(3): error: Inverses => true not compatible with GRevLex

It seems that selectVariables is not truly keeping the option "MonomialOrder=>Lex" in the subring QQ[x].

**Comment by DanGrayson** [comment]
Indeed, and in this case, it returns an answer, but the ordering is wrong:
```

i1 : R=QQ[x,y,z,MonomialOrder=>Lex];

i2 : describe first selectVariables({0,1},R)

o2 = QQ[x..y, Degrees => {2:1}, Heft => {1}, MonomialOrder => {MonomialSize => 32}, DegreeRank => 1]
                                                              {Lex => 0          }
                                                              {Position => Up    }
                                                              {GRevLex => {2:1}  }
```

**Comment by d-torrance** [comment]
> **Written by Claude** (Claude Opus 5, via Claude Code), not by @d-torrance, whose account posted it -- please weigh it accordingly.

Both transcripts above still reproduce byte-identically on 1.26.06-40-gd8e86d689d, and they are
the same bug: the wrong ordering @DanGrayson showed is what *causes* the error @MarthaBernal
reported.

### The Lex block always comes back empty

`selectVariables` rebuilds the monomial order by walking the blocks of `(options R).MonomialOrder`
left to right, using a shared `off` counter to remember where the current block starts.  Two
helpers do the counting, in `M2/Macaulay2/m2/polyrings.m2`:

```m2
pg := (v,wts) -> first(pw(v,wts), off = off + #wts)          -- line 250

pn := (v,nw) -> (
     off = off + nw;                                         -- line 252
     n:=0;
     for i in v do if i<off then continue else if i>=off+nw then break else n=n+1;
     n)
```

`pg` counts and *then* advances `off`; `pn` advances `off` and *then* counts.  So `pn` is
counting the selected variables that fall in the block after its own.  `pg` serves `GRevLex`,
which is why the default order has never shown this; `pn` serves `Lex`, `RevLex`, `GroupLex`,
`GroupRevLex` and `NCLex`, all of which do.

The consequence is not limited to selecting a subset.  Selecting *every* variable of a Lex ring
loses the order too:

```m2
i1 : R = QQ[x,y,z,MonomialOrder=>Lex];

i2 : describe first selectVariables({0,1,2},R)

o2 = QQ[x..z, Degrees => {3:1}, Heft => {1}, MonomialOrder => {MonomialSize => 32}]
                                                              {Lex => 0          }
                                                              {Position => Up    }
                                                              {GRevLex => {3:1}  }
```

`Lex => 0`, and all three variables picked up by the `GRevLex` clause that `fixup` appends for
the variables no block claimed.  That is a silently wrong monomial order, returned without any
error, which is the shape most likely to be hit without being noticed.

It is also where the reported error comes from: `Inverses => true` is not accepted against
`GRevLex`, so as soon as the variables fall through to that appended clause the monoid refuses
to be built.  Worth saying explicitly because the obvious reading of this issue — "the error
message is wrong" — invites a fix to the error rather than to the ordering, and that would leave
the wrong order in place everywhere it does not happen to error.

### Moving the increment

`selectVariables` is ordinary M2 code, so the correction can be checked without rebuilding, by
re-declaring the same definitions at top level with `off = off + nw` moved below the loop and
nothing else changed:

| ring | selection | now | with `off` advanced after the count |
| --- | --- | --- | --- |
| `QQ[x,y,z,MonomialOrder=>Lex]` | `{0,1}` | `Lex => 0`, `GRevLex => {2:1}` | `Lex => 2` |
| `QQ[x,y,z,MonomialOrder=>Lex]` | `{0,1,2}` | `Lex => 0`, `GRevLex => {3:1}` | `Lex => 3` |
| `QQ[x,y,MonomialOrder=>Lex,Inverses=>true]` | `{0}` | error | `GroupLex => 1`, builds |
| `QQ[a,b,c,d,MonomialOrder=>{Lex=>2,GRevLex=>2}]` | `{0,2}` | — | `Lex => 1`, `GRevLex => {1}` |
| `QQ[a,b,c,d]` | `{0,1}` | `QQ[a..b]` | unchanged |

The last two are the ones that keep a fix honest: the mixed-block case is what a correction that
simply dropped the increment would get wrong, and the plain `GRevLex` case confirms `pg`'s path
is untouched.

<sub>Triaged in a sweep of the open issues; source of truth: [`bug-triage/issues.tsv`](https://github.com/d-torrance/M2/blob/bug-triage/bug-triage/issues.tsv).</sub>
